include/ginkgo/core/base/types.hpp

```diff
@@ -546,12 +537,52 @@ GKO_ATTRIBUTES constexpr bool operator!=(precision_reduction x,
     _macro(std::complex<float>, std::complex<double>) GKO_NOT_IMPLEMENTED; \
     template <>                                                            \
     _macro(std::complex<double>, std::complex<float>) GKO_NOT_IMPLEMENTED
+
+
+#define GKO_INSTANTIATE_FOR_EACH_VALUE_CONVERSION_OR_COPY(_macro)         \
+    GKO_INSTANTIATE_FOR_EACH_VALUE_CONVERSION(_macro);                    \
+    template <>                                                           \
+    _macro(float, float) GKO_NOT_IMPLEMENTED;                             \
```

Comment:
Why is `float, float` also not implemented? I thought that `DPCPP_SINGLE` meant that only single precision would work, in which case this instantiation should be fine?

Reply:
Good catch, that was an oversight